vllm/v1/core/sched/scheduler.py

```diff
@@ -1494,8 +1494,10 @@ def _handle_invalid_blocks(self, invalid_block_ids: set[int]) -> set[str]:
         total_tokens_to_reschedule += num_tokens_to_reschedule
 
         # Mark requests with async KV load failures; they will be rescheduled
-        # once loading completes
+        # once loading completes. Also mark them as finished so they can
+        # transition back to WAITING status.
         self.failed_recving_kv_req_ids |= async_affected_req_ids
+        self.finished_recving_kv_req_ids |= async_affected_req_ids
```

Comment:
I don't think we want to do this. The contract is that all reqs will be returned via `get_finished()`, including ones with failures. We should make sure that the nixl connector still returns the req id in the finished recving set.

Reply:
thanks for this, turns out I was solving the infinite looping on the wrong side, moved handling back to nixl connector